src/main/java/uk/gov/hmcts/reform/pip/account/management/validation/validator/ValueOfEnumValidator.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.pip.account.management.validation.validator;
+
+import uk.gov.hmcts.reform.pip.account.management.validation.annotations.ValueOfEnum;
+
+import java.util.List;
+import java.util.stream.Stream;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * Validator class that checks if an input is in the supplied enum.
+ */
+public class ValueOfEnumValidator implements ConstraintValidator<ValueOfEnum, CharSequence> {
+    private List<String> acceptedValues;
+
+    @Override
+    public void initialize(ValueOfEnum annotation) {
+        acceptedValues = Stream.of(annotation.enumClass().getEnumConstants())
+            .map(Enum::name)
+            .toList();
+    }
+
+    @Override
+    public boolean isValid(CharSequence value, ConstraintValidatorContext context) {
```

Comment:
You could compress this into "return value == null || acceptedValues.contains(value.toString())"

Reply:
Fixed :)